app/src/components/RegexField.tsx

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useMemo, useState } from "react";
+import { FieldError, Input, Label } from "react-aria-components";
+import { fetchQuery, graphql, useRelayEnvironment } from "react-relay";
+import { debounce } from "lodash";
+
+import { RegexFieldQuery } from "@phoenix/components/__generated__/RegexFieldQuery.graphql";
+import { Text } from "@phoenix/components/content";
+import { TextField, TextFieldProps } from "@phoenix/components/field";
+import { Icon, Icons } from "@phoenix/components/icon";
+
+type RegexFieldProps = {
+  value: string;
+  onChange: (value: string) => void;
+  isInvalid?: boolean;
+  error?: string;
+  description?: string;
+  label?: string;
+  ariaLabel?: string;
+  placeholder?: string;
+} & Omit<TextFieldProps, "isInvalid" | "onChange">;
+
+export const RegexField = ({
+  value,
+  onChange,
+  error,
+  isInvalid,
+  description,
+  label,
+  ariaLabel,
+  placeholder,
+  ...textFieldProps
+}: RegexFieldProps) => {
+  return (
+    <TextField
+      isInvalid={isInvalid}
+      aria-label={ariaLabel || label}
+      value={value}
+      onChange={onChange}
+      {...textFieldProps}
+    >
+      {label && <Label>{label}</Label>}
+      <Input placeholder={placeholder} />
+      {!error && description && <Text slot="description">{description}</Text>}
+      {error && <FieldError>{error}</FieldError>}
+      {isInvalid && value && (
+        <Icon
+          style={{ color: "var(--ac-global-color-danger)" }}
+          svg={<Icons.CloseCircle />}
+        />
+      )}
+      {!isInvalid && value && (
+        <Icon
+          style={{ color: "var(--ac-global-color-success)" }}
+          svg={<Icons.CheckmarkCircleFilled />}
+        />
+      )}
+    </TextField>
+  );
+};
+
+const regexFieldQuery = graphql`
+  query RegexFieldQuery($input: String!) {
+    validateRegularExpression(regex: $input) {
+      isValid
+      errorMessage
+    }
+  }
+`;
```

Comment:
Just for my understanding - this is because I'd assume there's something inherently different between regex on the client and server but just trying to understand the difference to warrant the network call.